core/common/src/MapLayerSettings.ts

```diff
@@ -380,3 +346,112 @@ export class MapLayerSettings {
     return children;
   }
 }
+
+export interface BaseMapLayerProps extends MapLayerProps {
+  provider?: BackgroundMapProviderProps;
+}
+
+export class BaseMapLayerSettings extends MapLayerSettings {
+  private _provider?: BackgroundMapProvider;
+
+  public get provider(): BackgroundMapProvider | undefined { return this._provider; }
+
+  public static override fromJSON(props?: BaseMapLayerProps): BaseMapLayerSettings | undefined {
+    const settings = super.fromJSON(props);
+    if (!settings)
+      return undefined;
+
+    assert(settings instanceof BaseMapLayerSettings);
+    if (props?.provider)
+      settings._provider = BackgroundMapProvider.fromJSON(props.provider);
+
+    return settings;
+  }
+
+  public override toJSON(): BaseMapLayerProps {
+    const props = super.toJSON() as BaseMapLayerProps;
+    if (this.provider)
+      props.provider = this.provider.toJSON();
+
+    return props;
+  }
+
+  public override clone(changedProps: MapLayerProps): BaseMapLayerSettings {
+    const prevUrl = this.url;
+    const clone = super.clone(changedProps);
+    assert(clone instanceof BaseMapLayerSettings);

```

Comment:
Im seeing assertion failures here, clone appears to be an instance of MapLayerSettings and not BaseMapLayerSettings

Reply:
I have fixed that already. I am working on these 3 full stack tests.